### PR TITLE
Supports `debian_version` file content for `version(best=True)` calls

### PR DESCRIPTION
**Pull request by HorlogeSkynet**
Dear @python-distro/maintainers,

---

From ansible/ansible#77928 and #332, it appeared that Debian >= 9 do not ship anymore the `MINOR` SemVer segment in `lsb_release` "distribution release" output.
This means that setting `best=True` cannot be used to retrieve it, and thus `version(best=True)` behaves differently across Debian < 10 and Debian >= 10.

This patch implements `/etc/debian_version` file parsing to workaround this issue, on "Debian-like" distributions.

---

Many thanks, bye :wave:

### CODE REVIEW

File: tests/test_distro.py [hunk 1]
```diff
@@ -138,11 +138,10 @@ def _setup_for_distro(self, distro_root: str) -> None:
 class TestOSRelease:
     def setup_method(self, test_method: FunctionType) -> None:
         dist = test_method.__name__.split("_")[1]
-        os_release = os.path.join(DISTROS_DIR, dist, "etc", "os-release")
         self.distro = distro.LinuxDistribution(
             include_lsb=False,
-            os_release_file=os_release,
             distro_release_file="path-to-non-existing-file",
+            root_dir=os.path.join(DISTROS_DIR, dist),
```
Comment (hartwork) [inline tests/test_distro.py:144]: The intentions (and need) for the changes to `setup_method` above are not clear to me. Are they needed? Could you elaborate?

File: src/distro/distro.py [hunk 2]
```diff
@@ -898,6 +898,15 @@ def version(self, pretty: bool = False, best: bool = False) -> str:
         if self.uname_attr("id").startswith("aix"):
             # On AIX platforms, prefer oslevel command output.
             versions.insert(0, self.oslevel_info())
+        elif self.id() == "debian" or "debian" in self.like().split():
+            # On Debian-like, add debian_version file content to candidates list.
+            try:
+                with open(
+                    os.path.join(self.etc_dir, "debian_version"), encoding="ascii"
```
Comment (hartwork) [inline src/distro/distro.py:905]: This additional read^^ may interfere with #328, i.e. one of the two will need a rebase and additional changes after adjustments.

**Review by hartwork — COMMENTED** [review]
Hi @HorlogeSkynet I like the general direction :+1:, please see two comments on details below:

**Comment by HorlogeSkynet** [comment]
> Hi @HorlogeSkynet I like the general direction +1, please see two comments on details below:

Hey @hartwork ! Thanks for your quick feedback, see my replies to your comments :wink:

File: tests/test_distro.py (hunk 1 above)
Comment (HorlogeSkynet) [reply tests/test_distro.py:144]: It appeared that `TestOSRelease` test cases **were not** "chroot-ed" in their corresponding directories... Now that we open a file under `/etc` it became an issue (host's `/etc/debian_version` was read).
I've had to set `root_dir` to workaround this as well :confused:

File: src/distro/distro.py (hunk 2 above)
Comment (HorlogeSkynet) [reply src/distro/distro.py:905]: Definitely ! If we merge this beforehand, I'll perform a rebase for #328 :angel:

File: tests/test_distro.py (hunk 1 above)
Comment (hartwork) [reply tests/test_distro.py:144]: The access to `self.etc_dir`, I see! :+1:

**Review by hartwork — APPROVED** [review]

**Pull request merged**